### ISSUE TITLE
CopyQ will not capture PRIMARY (selection) on Wayland

### ISSUE THREAD

**Issue by hsantanna** [hluk/CopyQ#1644]
GTK 3.24.29 received a path that made PRIMARY work between QT/KDE apps on Wayland.
So we can mouse select text from (e.g) Firefox and than paste it by middle clicking to (e.g) Konsole.

But right now this text is not captured by CopyQ (git master/HEAD).

CopyQ will not capture PRIMARY (mouse selection), no matter if from GTK apps or QT apps.

**To Reproduce**
1. Start KDE Plasma (Wayland session)
2. Open any app (like a GTK app or a QT app);
3. Select text with mouse cursor;
4. Open another app;
5. Use middle click to paste the text previously selected, from the first app to the second one;
6. Check CopyQ, the selected (and pasted) text was not captured by CopyQ.

**Tested with:**
`QT_QPA_PLATFORM=wayland copyq`

Platform: Linux (Arch Linux)
QT: 5.15.2+kde+r180-1
Desktop: KDE Plasma 5.21.4 (Wayland)

**Comment by hluk** [comment]
CopyQ already supports selection in Wayland, using the available protocol: https://github.com/swaywm/wlr-protocols/blob/d1598e82240d6e8ca57729495a94d4e11d222033/unstable/wlr-data-control-unstable-v1.xml

I believe it's the compositor and apps (GUI frameworks and libs) that need to support it too on Wayland.

If this is broken in CopyQ, I will need more details.

**Comment by hluk** [comment]
Hmm, tested in Firefox in Sway, I can select text and paste with middle click other apps (and `wl-paste -p`). I can middle-click paste to CopyQ too but it cannot get the selection by itself for some reason.

```
$ WAYLAND_DEBUG=1 firefox-wayland
...
[3147518.183] gtk_primary_selection_device@42.selection(nil)
[3147518.212]  -> gtk_primary_selection_offer@4278190087.destroy()
[3147518.249] zwp_primary_selection_device_v1@18.selection(nil)
[3147518.283]  -> zwp_primary_selection_offer_v1@4278190086.destroy()
[3147518.337] gtk_primary_selection_device@42.data_offer(new id gtk_primary_selection_offer@3226289104)
[3147518.385] gtk_primary_selection_offer@4278190082.offer("text/html")
[3147518.424] gtk_primary_selection_offer@4278190082.offer("text/_moz_htmlcontext")
[3147518.464] gtk_primary_selection_offer@4278190082.offer("text/_moz_htmlinfo")
[3147518.504] gtk_primary_selection_offer@4278190082.offer("UTF8_STRING")
[3147518.541] gtk_primary_selection_offer@4278190082.offer("COMPOUND_TEXT")
[3147518.575] gtk_primary_selection_offer@4278190082.offer("TEXT")
[3147518.605] gtk_primary_selection_offer@4278190082.offer("STRING")
[3147518.642] gtk_primary_selection_offer@4278190082.offer("text/plain;charset=utf-8")
[3147518.675] gtk_primary_selection_offer@4278190082.offer("text/plain")
[3147518.712] gtk_primary_selection_offer@4278190082.offer("text/x-moz-url-priv")
[3147518.741] gtk_primary_selection_device@42.selection(gtk_primary_selection_offer@4278190082)
[3147518.776] zwp_primary_selection_device_v1@18.data_offer(new id zwp_primary_selection_offer_v1@3226289344)
[3147518.823] zwp_primary_selection_offer_v1@4278190083.offer("text/html")
[3147518.865] zwp_primary_selection_offer_v1@4278190083.offer("text/_moz_htmlcontext")
[3147518.931] zwp_primary_selection_offer_v1@4278190083.offer("text/_moz_htmlinfo")
[3147518.967] zwp_primary_selection_offer_v1@4278190083.offer("UTF8_STRING")
[3147519.008] zwp_primary_selection_offer_v1@4278190083.offer("COMPOUND_TEXT")
[3147519.049] zwp_primary_selection_offer_v1@4278190083.offer("TEXT")
[3147519.088] zwp_primary_selection_offer_v1@4278190083.offer("STRING")
[3147519.127] zwp_primary_selection_offer_v1@4278190083.offer("text/plain;charset=utf-8")
[3147519.164] zwp_primary_selection_offer_v1@4278190083.offer("text/plain")
[3147519.204] zwp_primary_selection_offer_v1@4278190083.offer("text/x-moz-url-priv")
[3147519.237] zwp_primary_selection_device_v1@18.selection(zwp_primary_selection_offer_v1@4278190083)
...
```

**Comment by tinywrkb** [comment]
@hluk https://bugzilla.mozilla.org/show_bug.cgi?id=1704043

**Comment by hluk** [comment]
I think the source application (e.g. Firefox) must support the `wlr_data_control_unstable_v1` protocol, otherwise CopyQ won't receive clipboard data when it is not focused.

**Comment by JosefLitos** [comment]
I think this is solved already, at least from my experience with FF on wayland the copied data is added to copyq

**Comment by hsantanna** [comment]
I can confirm as fixed. A GTK update solved the issue some time ago. I can't check which GTK version solved the issue, but right now, for GTK 3.24.38 its solved. 

Closing the issue.
